app/templates/account/payments-view.jade

```diff
@@ -10,28 +10,49 @@ block content
       a(href="/account", data-i18n="nav.account")
     li.active(data-i18n="account.payments")
 
-  if view.payments.size()
+  if view.prepaidMap
     table.table.table-striped
       tr
         th(data-i18n="account.purchased")
         th(data-i18n="account.paid_on")
-        th(data-i18n="account.service")
         th(data-i18n="account.price")
         th(data-i18n="account.gems")
       for payment in view.payments.models
-        - var service = payment.get('service')
+        - var service = payment.get('service') || '';
+        - var productID = payment.get('productID');
+        - var prepaidID = payment.get('prepaidID');
         tr
-          if payment.get('productID') === 'custom'
+          if prepaidID && view.prepaidMap[prepaidID]
+            case view.prepaidMap[prepaidID].get('type')
+              when 'subscription'
+                td(data-i18n='subscribe.stripe_description')
+              when 'terminal_subscription'
+                if view.prepaidMap[prepaidID].get('description')
+                  td= view.prepaidMap[prepaidID].get('description')
+                else
+                  td(data-i18n='subscribe.stripe_description')
+              when 'course'
+                td(data-i18n='special_offer.course_prefix')
+              when 'starter_license'
+                td(data-i18n='special_offer.student_starter_license')
+              default
+                td
+          else if productID === 'custom'
             td= payment.get('description')
-          else if payment.get('productID')
+          else if productID
             td(data-i18n='account.gems')
+          else if parseInt(payment.get('gems')) === 42000
+            if typeof(me.get('stripe').free) === 'boolean'
+              if me.get('stripe').customerID
+                td(data-i18n='subscribe.lifetime')
+              else
+                td(data-i18n='subscribe.free_subscription')
+            else
+              td(data-i18n='subscribe.year_subscription')
```

Comment:
Lines 44-51 feel very fragile to changes, especially the gems part. Is there no better way to determine what kind of subscription they have?